src/SFA.DAS.PR.Application/AccountProviderLegalEntities/Queries/GetAccountProviderLegalEntities/GetAccountProviderLegalEntitiesQueryValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using FluentValidation;
+
+namespace SFA.DAS.PR.Application.AccountProviderLegalEntities.Queries.GetAccountProviderLegalEntities;
+public class GetAccountProviderLegalEntitiesQueryValidator : AbstractValidator<GetAccountProviderLegalEntitiesQuery>
+{
+    public const string UkprnAccountHashIdValidationMessage = "Either one of Ukprn or AccountHashId is required.";
+    public const string UkprnFormatValidationMessage = "Currently a Ukprn must start with the value 1 and should be 8 digits long.";
+    public const string OperationFilterValidationMessage = "Currently at least one operation filter must to be supplied.";
+    public const string OperationFilterFormatValidationMessage = "Operation filter values are limited to 0, 1 or 2.";
+    public GetAccountProviderLegalEntitiesQueryValidator()
+    {
+        RuleFor(x => x.Ukprn)
+            .Must(Ukprn => Ukprn.HasValue)
+            .WithMessage(UkprnAccountHashIdValidationMessage)
+            .When(x => string.IsNullOrEmpty(x.AccountHashedId));
+
+        RuleFor(x => x.AccountHashedId)
+            .Must(AccountHashedId => !string.IsNullOrWhiteSpace(AccountHashedId))
+            .WithMessage(UkprnAccountHashIdValidationMessage)
+            .When(x => !x.Ukprn.HasValue);
+
+        RuleFor(model => model.Ukprn)
```

Comment:
This should be chained with NotEmpty if possible to keep things together. Also we need to start looking at reusability of these validations.

Reply:
Spoke to Mark about the reusability, as either Ukprn or PublicHashId can be null for this specific endpoint the validators cannot be reused for his specific scenario.